### PR TITLE
Add new shipping class modal to a shipping class section in product page

### LINKED ISSUE #34657: [Shipping] Shipping class dropdown add new option
**User story**

- As a merchant, I want to the ability to create new shipping classes within the product edit screen.

**Design**
(design: `5sAIeTRd9Yp7nSCT33BAWz-fi-6590%3A287898`)

**Acceptance criteria**

- [x] A new `New shipping class` option should be added to the dropdown
- [x] When the New shipping class option is selected, we show a modal where the user can enter the name, slug, and description 
- [x] The name and slug are automatically populated based on the first added product category. For example, if the product is added to `Shirts`, the class’s name and slug will be `Shirts`.
- [x] If the product doesn’t have any categories yet, the name and slug are respectively populated with `New shipping class` and `new-shipping-class`.
- [x] When created, the name of the new shipping class appears in the dropdown.

### PR DESCRIPTION
**Pull request by mdperez86**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/34657.

### How to test the changes in this Pull Request:

1. Go to Add new product page.
2. Under Shipping Class section click on the Shipping class dropdown, an `Add new shipping class` option should be shown as the second option of the list.
3. When click `Add new shipping class` option a modal to add a new shipping class should be shown.
4. If the product does not have a category then the fields `Name` and `Slug` on the modal should be filled with `New shipping class` and `new-shipping-class` values respectively. 

<video src="https://user-images.githubusercontent.com/13334210/193664226-e1a83c33-8129-4d20-b2c7-2f7e06fd2bf0.mov"></video>
5. If the product contains a valid category different than `Uncategorized` then that value should be used to fill the fields `Name` and `Slug` on the modal. 

<video src="https://user-images.githubusercontent.com/13334210/193664590-cfc4f1ed-453a-499a-9f98-fdc3e2783d2e.mov"></video>
6. After adding the new shipping class the dropdown behind the modal should be updated with the new option added.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you successfully run tests with your changes locally?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> run changelog add`?

### FOR PR REVIEWER ONLY:

-   [x] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by pmcpinto** [comment]
@mdperez86 overall, the flow looks good to me. I just have two comments:
- It would be good to inform users that the shipping class will be global (available to any product)
- When the user creates a shipping class: I feel that we should automatically add the product to this class rather than having the user open the dropdown again. 
@jarekmorawski what do you think?

**Comment by AnnaMag** [comment]
> It would be good to inform users that the shipping class will be global (available to any product)
> When the user creates a shipping class: I feel that we should automatically add the product to this class rather than having the user open the dropdown again.

Let's keep the scope of the PR and open a new issue for the follow-up changes. cc @jarekmorawski

**Review by AnnaMag — DISMISSED** [review]
Tests well. In agreement with the initial AC.
Thanks!

File: packages/js/components/src/form/form.tsx [hunk 1]
```diff
@@ -262,7 +262,7 @@ function FormComponent< Values extends Record< string, any > >(
 			}
 
 			if ( callback ) {
-				callback( values );
+				return callback( values );
```
Comment (AnnaMag) [inline packages/js/components/src/form/form.tsx:265]: Why this change? 
Splitting the changes into more than 1 commit and making the commits bit more more descriptive, would make the review easier.

Comment (mdperez86) [reply packages/js/components/src/form/form.tsx:265]: > Why this change?

When submitting the form we need a feedback in order to complete other things like dealing with server error or success mesages. With this we can do something like `onSubmit().then(...).catch(...)` to handle post submit tasks.

 > Splitting the changes into more than 1 commit and making the commits bit more more descriptive, would make the review easier.

You are right! I'll do this always from now. Thanks!!!

Comment (AnnaMag) [reply packages/js/components/src/form/form.tsx:265]: Thanks for the answer!

**Review by AnnaMag — APPROVED** [review]
LGTM
